crates/rpc/src/handlers/state.rs

```diff
@@ -1,86 +1,163 @@
-    )))?;
+    .map_err(error::ErrorInternalServerError)?
+    .ok_or_else(|| {
+        error::ErrorNotFound(format!("Failed to find `block_root` from {state_id:?}"))
+    })?;
 
     db.beacon_state_provider()
         .get(block_root)
-        .map_err(|_| ApiError::InternalError)?
-        .ok_or(ApiError::NotFound(format!(
-            "Failed to find `beacon_state` from {block_root:?}"
-        )))
+        .map_err(error::ErrorInternalServerError)?
+        .ok_or_else(|| {
+            error::ErrorNotFound(format!("Failed to find `block_root` from {state_id:?}"))
+        })
 }
 
-pub async fn get_state(state_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let state = get_state_from_id(state_id, &db).await?;
+#[get("/beacon/states/{state_id}")]
+pub async fn get_beacon_state(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
 
-    Ok(with_status(BeaconResponse::json(state), StatusCode::OK))
+    Ok(HttpResponse::Ok().json(BeaconResponse::from(state)))
 }
 
-pub async fn get_state_root(state_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let state = get_state_from_id(state_id, &db).await?;
+#[get("/beacon/states/{state_id}/root")]
+pub async fn get_state_root(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
 
     let state_root = state.tree_hash_root();
 
-    Ok(with_status(
-        BeaconResponse::json(RootResponse::new(state_root)),
-        StatusCode::OK,
-    ))
+    Ok(HttpResponse::Ok().json(BeaconResponse::from(state_root.to_string())))
 }
+
+/// Called by `/eth/v1/beacon/states/{state_id}/fork` to get fork of state.
+#[get("/beacon/states/{state_id}/fork")]
+pub async fn get_state_fork(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
+
+    Ok(HttpResponse::Ok().json(state.fork))
+}
+
+/// Called by `/states/<state_id>/finality_checkpoints` to get the Checkpoint Data of state.
+#[get("/beacon/states/{state_id}/finality_checkpoints")]
+pub async fn get_state_finality_checkpoint(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
+
+    let response = CheckpointData::new(
+        state.previous_justified_checkpoint,
+        state.current_justified_checkpoint,
+        state.finalized_checkpoint,
+    );
+    Ok(HttpResponse::Ok().json(response))
```

Comment:
```suggestion
#[get("/beacon/states/{state_id}/finality_checkpoints")]
pub async fn get_state_finality_checkpoint(
    db: web::Data<ReamDB>,
    state_id: web::Path<ID>,
) -> actix_web::Result<impl Responder> {
    let state = get_state_from_id(state_id.into_inner(), &db)?;

    Ok(HttpResponse::Ok().json(BeaconResponse::new(CheckpointData::new(
        state.previous_justified_checkpoint,
        state.current_justified_checkpoint,
        state.finalized_checkpoint,
    ))))
```

Here is an example, can you review your PR and check to see if you made any other regressions like this and fix them, then ping me when it is done and I will review it